src/test/java/com/connexta/ingest/IngestITests.java

```diff
@@ -139,55 +269,100 @@ public void testIngestRequestWithMissingMetacard() throws Exception {
-
-    mvc.perform(
-            multipart("/ingest")
-                .file("file", TEST_FILE)
-                .file("metacard", TEST_METACARD)
-                .param("correlationId", "000f4e4a")
-                .header("Accept-Version", "1.2.1")
-                .accept(MediaType.APPLICATION_JSON)
-                .contentType(MediaType.MULTIPART_FORM_DATA))
-        .andExpect(status().isInternalServerError());
+    transformMockRestServiceServer.expect(never(), requestTo(endpointUrlTransform));
+
+    webTestClient
+        .post()
+        .uri("/ingest")
+        .header("Accept-Version", "1.1.1")
+        .contentType(MediaType.MULTIPART_FORM_DATA)
+        .accept(MediaType.APPLICATION_JSON)
+        .syncBody(TEST_INGEST_REQUEST_BODY)
+        .exchange()
+        .expectStatus()
+        .is5xxServerError();
   }
 
-  /* END store request tests */
+  /* END store file request tests */
+
+  // TODO testUnsuccessfulStoreMetacardRequest
 
   /* START transform request tests */
 
-  // The error handler throws the same exception for all non-202 status codes returned by the
-  // transformation endpoint.
+  /**
+   * The error handler throws the same exception for all non-202 status codes returned by the
+   * transformation endpoint.
+   */
   @Test
   public void testUnsuccessfulTransformRequest() throws Exception {
     final URI location = new URI("http://localhost:1232/store/1234");
-    storeServer
+    storeMockRestServiceServer
         .expect(requestTo(endpointUrlStore))
         .andExpect(method(HttpMethod.POST))
         .andRespond(withCreatedEntity(location));
 
-    transformServer
+    transformMockRestServiceServer
         .expect(requestTo(endpointUrlTransform))
         .andExpect(method(HttpMethod.POST))
         .andExpect(header("Accept-Version", endpointsTransformVersion))
         .andExpect(jsonPath("$.location").value(location.toString())) // TODO assert URI, not String
+        .andExpect(jsonPath("$.mimeType").value(TEST_FILE_CONTENT_TYPE))
+        .andExpect(jsonPath("$.metacardLocation").isNotEmpty())
         .andRespond(withServerError());
 
-    mvc.perform(
-            multipart("/ingest")
-                .file("file", TEST_FILE)
-                .file("metacard", TEST_METACARD)
-                .param("correlationId", "000f4e4a")
-                .header("Accept-Version", "1.2.1")
-                .accept(MediaType.APPLICATION_JSON)
-                .contentType(MediaType.MULTIPART_FORM_DATA))
-        .andExpect(status().isInternalServerError());
+    webTestClient
+        .post()
+        .uri("/ingest")
+        .header("Accept-Version", "1.1.1")
+        .contentType(MediaType.MULTIPART_FORM_DATA)
+        .accept(MediaType.APPLICATION_JSON)
+        .syncBody(TEST_INGEST_REQUEST_BODY)
+        .exchange()
+        .expectStatus()
+        .is5xxServerError();
   }
+
   /* END transform request tests */
+
+  @NotNull
+  private static Matcher<Resource> hasContents(final InputStream expected) {
```

Comment:
:confetti_ball: This is great! I wish we had place to keep reusable code like this.